src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -257,27 +307,34 @@ internal class ModelCheckingStrategy(
             }
 
         fun isSwitchPosition() = executionPosition in switchPositions
+        fun isCrashPosition() = executionPosition in crashPositions
 
         /**
-         * Creates a new execution position that corresponds to the current switch point.
+         * Creates a new execution position that corresponds to the current switch/crash point.
          * Unlike switch points, the execution position is just a gradually increasing counter
          * which helps to distinguish different switch points.
          */
-        fun newExecutionPosition(iThread: Int) {
+        private fun newExecutionPosition(iThread: Int, isSwitch: Boolean) {
             executionPosition++
-            if (executionPosition > switchPositions.lastOrNull() ?: -1) {
+            if (exploreCrashNode == isSwitch) return
+            if (executionPosition > positions.lastOrNull() ?: -1) {
```

Comment:
Instead of creating `positions`, you can compare here with the last execution event position (maximum of `crashPositions.last()` and `switchPositions.last()`). Probably this logic can be extracted to `lastChosenExecutionPosition()` method.

Reply:
`positions` does not create a copy of a list, it is just a link. 
I think comparing with maximum does not work. For example, we are exploring a crash node and our crashPositions list is empty while switchPositions is [42]. We are visiting the current crash position 33, so we want to add it while it is smaller than 42.

Reply:
If `crashPositions` is empty and `switchPositions` is not, you are interested only in crash positions that happen after the last event (switch). Adding a crash before the last switch can break the switch choice (e.g., when a thread you chose to switch to becomes no longer available)

Reply:
You are right. I fixed it